### PR TITLE
Add Phase of Moon app

### PR DESCRIPTION
**Pull request by alan-fleming**
Displays an image and a text description of the current phase of the moon. Accurate in both hemispheres.

phase_of_moon

### CODE REVIEW

**Comment by alan-fleming** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by henryso** [comment]
recheck

File: apps/phaseofmoon/phase_of_moon.star [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                    render.Image(src=base64.decode(phaseImage)),
+                    render.Padding(
+                        pad=(1,0,0,0),
+                        child =  render.WrappedText(moonPhase),
+                    )
+                ]
+            )
+        )
+    )
+
```
Comment (betterengineering) [inline apps/phaseofmoon/phase_of_moon.star:116]: You will need this schema field to ensure we populate the `location` config object in `location = config.get("location")`:

```suggestion

def get_schema():
    return schema.Schema(
        version = "1",
        fields = [
            schema.Location(
                id = "location",
                name = "Location",
                desc = "Location for which to display the moon phase for.",
                icon = "place",
            ),
        ],
    )
```

**Review by betterengineering — APPROVED** [review]
Thanks for putting this together @alan-fleming, this looks great! I love that it's purely in starlark and takes the hemispheres into account ❤️ . You will need the schema added but otherwise this is good to go.

**Comment by betterengineering** [comment]
You can run the following to fix the linter issues locally, then you can commit the changes:
```
make format
```

**Comment by henryso** [comment]
I love this kind of app...

One thing, you don't need to project the time into any particular time zone to get `unix` out of it as UNIX time is based on the number of seconds from 1 Jan 1970 00:00:00 UTC, which is the same regardless of timezone.

File: apps/phaseofmoon/phase_of_moon.star (hunk 1 above)
Comment (alan-fleming) [reply apps/phaseofmoon/phase_of_moon.star:116]: How did I forget that? I even had the module added! On its way.

**Comment by alan-fleming** [comment]
> I love this kind of app...
> 
> One thing, you don't need to project the time into any particular time zone to get `unix` out of it as UNIX time is based on the number of seconds from 1 Jan 1970 00:00:00 UTC, which is the same regardless of timezone.

Thanks - v new at this, so I'll amend here and that will come up in a future push

**Comment by alan-fleming** [comment]
Does anyone have any suggestions as to why the build is failing? I'm outside my expertise here.

**Comment by henryso** [comment]
@alan-fleming There's a blank line at then end of the star file.  Delete that line.

**Comment by alan-fleming** [comment]
@henryso I'm absolutely certain the previous build failed because the linter complained there _wasn't_ a blank line at the end!
Oh well... learning is fun...

**Comment by rs7q5** [comment]
I think when I ran make lint on my system before committing it adds the line. I noticed last night when I ran `make lint` it adds it, so it may not like two empty lines?

**Pull request merged**